tests/Integration/Behaviour/Features/Scenario/Product/Combination/combinations_listing.feature

```diff
@@ -0,0 +1,99 @@
+      | reference      | combination name        | attributes           | impact on price | final price | quantity | is default |
+      | product1Blue   | Size - S, Color - Blue  | [Size:S,Color:Blue]  | 0               | 0           | 0        | false      |
+      | product1MWhite | Size - M, Color - White | [Size:M,Color:White] | 0               | 0           | 0        | false      |
+    And I should see following combinations of product product1 in page 3 limited to maximum 2 per page:
+      | reference      | combination name        | attributes           | impact on price | final price | quantity | is default |
+      | product1MBlack | Size - M, Color - Black | [Size:M,Color:Black] | 0               | 0           | 0        | false      |
+      | product1MBlue  | Size - M, Color - Blue  | [Size:M,Color:Blue]  | 0               | 0           | 0        | false      |
+    And there should be no combinations of product1 in page 4 when limited to maximum 2 per page
+    And I should see following combinations of product product1 in page 2 limited to maximum 3 per page:
+      | reference      | combination name        | attributes           | impact on price | final price | quantity | is default |
+      | product1MWhite | Size - M, Color - White | [Size:M,Color:White] | 0               | 0           | 0        | false      |
+      | product1MBlack | Size - M, Color - Black | [Size:M,Color:Black] | 0               | 0           | 0        | false      |
+      | product1MBlue  | Size - M, Color - Blue  | [Size:M,Color:Blue]  | 0               | 0           | 0        | false      |
+    And there should be no combinations of product1 in page 3 when limited to maximum 3 per page
+
+  Scenario: Product combinations filtering by attributes returns correct results
+    Given product product1 should have following list of combinations:
+      | reference      | combination name        | attributes           | impact on price | final price | quantity | is default |
+      | product1SWhite | Size - S, Color - White | [Size:S,Color:White] | 0               | 0           | 0        | true       |
+      | product1SBlack | Size - S, Color - Black | [Size:S,Color:Black] | 0               | 0           | 0        | false      |
+      | product1Blue   | Size - S, Color - Blue  | [Size:S,Color:Blue]  | 0               | 0           | 0        | false      |
+      | product1MWhite | Size - M, Color - White | [Size:M,Color:White] | 0               | 0           | 0        | false      |
+      | product1MBlack | Size - M, Color - Black | [Size:M,Color:Black] | 0               | 0           | 0        | false      |
+      | product1MBlue  | Size - M, Color - Blue  | [Size:M,Color:Blue]  | 0               | 0           | 0        | false      |
+    Then I should see following combinations of product product1 in page 1 limited to maximum 10 per page and filtered by attributes "S,M":
```

Comment:
Not sure about this, its quite long string, if we add more filters its gonna get unreadable. :(
Don't have any other ideas though

Reply:
```suggestion
    When I paginate product combinations with 10 items per page and I filter on attributes "S,M"
    Then page 1 should contain
```

We could create intermediate steps for building the query? This will need the introduction of a property `$productCombinationFilterSettings` in the Context

Reply:
yep the only option.
Altough I know someone does not like that, because I have done that with the pagination when i just started this feature

Reply:
related https://github.com/PrestaShop/PrestaShop/pull/20518#discussion_r530943495
@jolelievre maybe we should use something like that in this case now that its getting more complicated?

Reply:
I like your idea @matks

Reply:
I'm not a fan of splitting tests like this indeed because they become stateful and it can very easily lead to mistakes if we don't pay attention (false positive, and false negative) but I don't see a better alternative

If we could have two `TableNode` parameters it would be perfect!!

Reply:
One detail about the `attributes "S,M"` one thing I'm worried is that it only contains the attribute value but not its group What if you want to filter with two different groups `attributes "S,white"`

It will work as long as we don't have same names in different groups I guess, maybe I'm overthinking (our attributes are limited in behat anyway) but should we have a syntax that involves the group to be sure like `attributes "size:S,size:M,color:red"`

Reply:
> If we could have two TableNode parameters it would be perfect!!

tried, doesn't work :( it can only have one argument